src/sdl2_wrapper.cpp

```diff
@@ -771,6 +780,66 @@ CAMLprim value Val_SDL_Event(SDL_Event *event) {
 
     Store_field(v, 0, vInner);
     break;
+  case SDL_DROPTEXT:
+    v = caml_alloc(1, 25);
+    vInner = caml_alloc(5, 0);
+
+    getNonFocusedMousePosition(SDL_GetWindowFromID(event->drop.windowID), &mousePosX, &mousePosY);
+
+    Store_field(vInner, 0, Val_int(event->drop.windowID));
+    Store_field(vInner, 1, Val_some(caml_copy_string(event->drop.file)));
+    Store_field(vInner, 2, Val_int(event->drop.timestamp));
+    Store_field(vInner, 3, Val_int(mousePosX));
+    Store_field(vInner, 4, Val_int(mousePosY));
+
+    Store_field(v, 0, vInner);
+    SDL_free(event->drop.file);
+    break;
+  case SDL_DROPFILE:
+    v = caml_alloc(1, 26);
+    vInner = caml_alloc(5, 0);
+
+    getNonFocusedMousePosition(SDL_GetWindowFromID(event->drop.windowID), &mousePosX, &mousePosY);
+
+    Store_field(vInner, 0, Val_int(event->drop.windowID));
+    Store_field(vInner, 1, Val_some(caml_copy_string(event->drop.file)));
+    Store_field(vInner, 2, Val_int(event->drop.timestamp));
+    Store_field(vInner, 3, Val_int(mousePosX));
+    Store_field(vInner, 4, Val_int(mousePosY));
+
+    Store_field(v, 0, vInner);
+    SDL_free(event->drop.file);
+    break;
+  case SDL_DROPBEGIN:
+    v = caml_alloc(1, 27);
+    vInner = caml_alloc(5, 0);
+
+    getNonFocusedMousePosition(SDL_GetWindowFromID(event->drop.windowID), &mousePosX, &mousePosY);
+
+    Store_field(vInner, 0, Val_int(event->drop.windowID));
+    Store_field(vInner, 1, Val_none);
+    Store_field(vInner, 2, Val_int(event->drop.timestamp));
+    Store_field(vInner, 3, Val_int(mousePosX));
+    Store_field(vInner, 4, Val_int(mousePosY));
+
+    Store_field(v, 0, vInner);
+    SDL_free(event->drop.file);
+    break;
+  case SDL_DROPCOMPLETE:
+    v = caml_alloc(1, 28);
+    vInner = caml_alloc(5, 0);
+
+    getNonFocusedMousePosition(SDL_GetWindowFromID(event->drop.windowID), &mousePosX, &mousePosY);
+
+    Store_field(vInner, 0, Val_int(event->drop.windowID));
+    Store_field(vInner, 1, Val_none);
+    Store_field(vInner, 2, Val_int(event->drop.timestamp));
+    Store_field(vInner, 3, Val_int(mousePosX));
+    Store_field(vInner, 4, Val_int(mousePosY));
+
+    Store_field(v, 0, vInner);
+    SDL_free(event->drop.file);
+    break;
```

Comment:
Maybe I'm missing something here, (which is entirely possible since I've not exactly done much in this repo).

But can these 4 cases be dropped to 2 cases with an if statement at the start (like the mouse down/up events).
If I'm being dumb, just ignore/say so! I could be missing something, but right now it looks like there is two duplicated bits with just a different tag in the `caml_alloc`. Could potentially go even further and squash to one, but that seems a bit messy.

Obviously doesn't change functionality at all, just thinking about if this gets changed at any point.

Reply:
You're definitely right. Not sure why I didn't think of that. Will fix that in a sec.

Reply:
My preference would be to factor out the common code to a function instead. That kind of special casing quickly leads to hairy spaghetti code as requirements change and the code evolves. Abstractions like this are what functions are for after all.